src/pages/iou/IOUTransactions.js

```diff
@@ -17,33 +18,105 @@ const propTypes = {
 
     /** ReportID for the associated IOU report */
     iouReportID: PropTypes.number.isRequired,
+
+    /** Email for the authenticated user */
+    userEmail: PropTypes.string.isRequired,
+
+    /** Has the iouReport been paid? */
+    isReportPaid: PropTypes.bool,
 };
 
 const defaultProps = {
     reportActions: {},
+    isReportPaid: false,
 };
 
-const IOUTransactions = ({
-    reportActions,
-    chatReportID,
-    iouReportID,
-}) => (
-    <View style={[styles.mt3]}>
-        {_.map(reportActions, (reportAction) => {
-            if (reportAction.actionName === 'IOU'
-                && reportAction.originalMessage.IOUReportID === iouReportID) {
-                return (
-                    <ReportTransaction
-                        chatReportID={chatReportID}
-                        iouReportID={iouReportID}
-                        action={reportAction}
-                        key={reportAction.sequenceNumber}
-                    />
-                );
+class IOUTransactions extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            rejectableTransactionIDs: [],
+        };
+
+        this.updateRejectableTransactions = this.updateRejectableTransactions.bind(this);
+    }
+
+    componentDidMount() {
+        this.updateRejectableTransactions();
+    }
+
+    componentDidUpdate(oldProps) {
+        if (oldProps.reportActions !== this.props.reportActions) {
+            this.updateRejectableTransactions();
+        }
+    }
+
+    /**
+     * Update the rejectableTransactionIDs state array. A transaction must meet multiple requirements in order to be
+     * rejectable. We must exclude transactions not associated with the iouReportID, actions which have already been
+     * rejected, and those which are not of type 'create').
+     */
+    updateRejectableTransactions() {
+        if (this.props.isReportPaid) {
+            this.setState({rejectableTransactionIDs: []});
```

Comment:
NAB, this method could be changed to something like `updateRejectableTransactions()` and just called from the render method to modify something like `this.rejectableTransactionIDs`. Putting it on `state` like we are doing now also works, but doesn't seem like an optimization we need to do or think about.

Reply:
Yeah, that makes sense. I've removed the early optimization and will now call it from render.